### PR TITLE
Port our WebDAV content provider from serf/apr/apr-util, to curl.

### PR DESCRIPTION
**Pull request by DamjanJovanovic**
Remove serf/apr/apr-from the build.
Curl must now use OpenSSL, and linking to zlib helps.
See the email on https://lists.apache.org/thread/0k8vnbcr0z7r2boftz4zdq9r69w6n8cc for details.

Patch by: me

### CODE REVIEW

File: main/ucb/source/ucp/webdav/CurlRequest.cxx [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ *************************************************************/
+
+
+
+// MARKER(update_precomp.py): autogen include statement, do not remove
+#include "precompiled_webdav.hxx"
+
```
Comment (ardovm) [inline main/ucb/source/ucp/webdav/CurlRequest.cxx:26]: Please include string.h
Otherwise the compiler cannot find memcpy(3).

**Review by ardovm — CHANGES_REQUESTED** [review]
One small edit is necessary in order to build with our CentOS Docker image

File: main/ucb/source/ucp/webdav/CurlRequest.cxx (hunk 1 above)
Comment (DamjanJovanovic) [reply main/ucb/source/ucp/webdav/CurlRequest.cxx:26]: Done.

**Review by ardovm — APPROVED** [review]

**Comment by ardovm** [comment]
I just opened https://github.com/DamjanJovanovic/openoffice/pull/1 with my proposal for using the bundled OpenSSL under Unix.

**Pull request merged**